compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -420,7 +416,7 @@ class DottyBackendInterface(val outputDirectory: AbstractFile, val superCallsMap
      *  which we represent as classes.
      */
     def isEmittedInterface: Boolean = isInterface ||
-      sym.is(Flags.JavaDefined) && (toDenot(sym).isAnnotation || sym.is(Flags.ModuleClass) && symHelper(sym.companionClass).isInterface)
+      sym.is(Flags.JavaDefined) && (toDenot(sym).isAnnotation || sym.is(Flags.ModuleClass) && (sym.is(Flags.PureInterface)) || sym.is(Flags.Trait))
```

Comment:
I don't think that inlining is correct, the call to isInterface was on sym.companionClass, not sym

Reply:
Indeed. Good catch. Fixed it.